### PR TITLE
[Az.Attestation] Improved error messages and updated examples in documentation

### PR DESCRIPTION
**Pull request by gkostal**
## Description
Based on user feedback, updated Az.Attestation PowerShell:
- With improved error messages when service responds with Http Status Code 400 or 401
- With accurate and consistent examples in the documentation
- Renamed VSMEnclave to VBSEnclave

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Attestation/Attestation/Models/AttestationDataServiceClient.cs [hunk 1]
```diff
@@ -163,10 +164,28 @@ private string GetDataPlaneUri(string name, string resourceGroupName, bool refre
...
+                    {
+                        var error = JsonConvert.DeserializeObject<ServerError>(responseBody).Error;
+                        errorDetails += $"\n\rCode: {error.Code}\n\rMessage: {error.Message}\n\r";
+                    }
+                    catch (Exception)
+                    {
+                        errorDetails += $"\n\rResponse Body: {responseBody}\n\r";
+                    }
+                }
+                throw new RestException(errorDetails);
```
Comment (VeryEarly) [inline src/Attestation/Attestation/Models/AttestationDataServiceClient.cs]: I suppose test case need to be updated if there is one and is expecting the "ArgumentException"

File: src/Attestation/Attestation/help/New-AzAttestation.md [hunk 2]
```diff
@@ -192,6 +192,8 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
 
 ### System.String
 
+### System.Collections.Hashtable
```
Comment (VeryEarly) [inline src/Attestation/Attestation/help/New-AzAttestation.md]: nice catch

File: src/Attestation/Attestation/Models/AttestationDataServiceClient.cs (hunk 1 above)
Comment (gkostal) [reply src/Attestation/Attestation/Models/AttestationDataServiceClient.cs]: Good question.  There are no test cases specifically looking for the value "ArgumentException".  All tests are passing.

File: src/Attestation/Attestation/help/New-AzAttestation.md (hunk 2 above)
Comment (gkostal) [reply src/Attestation/Attestation/help/New-AzAttestation.md]: :-)  Thanks.  Just running the ```UpdateMarkdownHelp``` command with the ```UpdateInputOutput``` parameter.  [Doc link.](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
>If you would like to update the inputs/outputs for a markdown file, please run this cmdlet with the -UpdateInputOutput parameter.

**Pull request merged**